punchcut/src/scaler.rs

```diff
@@ -0,0 +1,177 @@
+use super::{source::*, Context, Error, Hinting, NormalizedCoord, Outline, Result, Variation};
+
+use read_fonts::{
+    types::{Fixed, GlyphId, Tag},
+    TableProvider,
+};
+
+use core::{borrow::Borrow, str::FromStr};
+
+/// Builder for configuring a glyph scaler.
+pub struct ScalerBuilder<'a> {
+    context: &'a mut Context,
+    font_id: Option<u64>,
+    size: f32,
+    hint: Option<Hinting>,
+}
+
+impl<'a> ScalerBuilder<'a> {
+    /// Creates a new builder for configuring a scaler with the given context.
+    pub fn new(context: &'a mut Context) -> Self {
+        context.coords.clear();
+        context.variations.clear();
+        Self {
+            context,
+            font_id: None,
+            size: 0.0,
+            hint: None,
+        }
+    }
+
+    /// Sets a unique font identifier for hint state caching.
+    pub fn font_id(mut self, font_id: Option<u64>) -> Self {
```

Comment:
do we expect the user to call this with `None`? Who is responsible for generating these ids?

Reply:
The choice of allowing `None` here is to allow builder methods to be called with the "default" values to avoid breaking the method chain.

The client would be responsible for generating the ids. I tried with a typed auto-generated unique id (CacheKey) in swash but the friction was real when integrating with other code so I wanted to make this as generic as possible.